src/main/java/org/tarantool/TarantoolClientConfig.java

```diff
@@ -48,4 +48,10 @@
     public long initTimeoutMillis = 60 * 1000L;
     public long writeTimeoutMillis = 60 * 1000L;
 
+    /**
+     * Connection establishment policies.
+     */
+    public int connectionTimeout = 2 * 1000;
+    public int retryCount = 3;
```

Comment:
Whether it is applied to one connection in case of cluster client?

Reply:
No, and this is a reason why I marked this PR as WIP. Now, retryCount is just the total number of attempts to connect. It may be a source of unreachable provider addresses for a RR socket provider. Say, `retryCount=3` and RR addresses pool contains 5 addresses. Probably, the last two may be unused in some circumstances.
On the other hand, use `retryCount` per one RR-address is quite unclear. What should the cluster client do when `retryCount=-1`. Or what value should be taken as total attempts number in this case?

Reply:
I would consider -1 as incorrect value (and raise an exception) for the cluster client. Another cluster-client specific option can control overall attempts count (but I doubt it is really needed).

I judge from the opinion that those options are usually set based on a visible network conditions like 'a connection usually succeeds after N attempts with M interval between them, otherwise we can consider an instance as down'. (`connectionTimeout` works here as the interval.)

Reply:
Okay, RR provider follows the _visible network conditions_ as mentioned above. Overall count was chosen as a number of addresses kept inside the provider. One attempt to obtain a channel equals using one address. In this way, after `pool.size` attempts the provider goes through all instances unsuccessfully and there are no more reasons to repeat it again.

Reply:
This approach looks okay for me.